micromagneticdata/tests/test_combined_drive.py

```diff
@@ -0,0 +1,138 @@
+import itertools
+import os
+
+import discretisedfield as df
+import numpy as np
+import pytest
+import ubermagtable as ut
+import xarray as xr
+
+import micromagneticdata as md
+
+
+class TestDrive:
+    # drives 0, 1, 2, 4: TimeDriver
+    # drives 3, 5: MinDriver
+    # drives 6: HysteresisDriver
+    def setup(self):
+        self.dirname = os.path.join(os.path.dirname(__file__), "test_sample")
+        self.name = "system_name"
+        self.data = md.Data(name=self.name, dirname=self.dirname)
+        self.combined_drives = [
+            self.data[0] << self.data[1] << self.data[2],
+            self.data[3] << self.data[5],
+            self.data[6] << self.data[6],
+        ]
+
+    def test_init(self):
+        combined_drive = md.CombinedDrive(self.data[0], self.data[1])
+        assert isinstance(combined_drive, md.CombinedDrive)
+        with pytest.raises(ValueError):
+            md.CombinedDrive(self.data[0])
+        with pytest.raises(TypeError):
+            md.CombinedDrive("wrong type")
+
+    def test_repr(self):
+        for combined in self.combined_drives:
+            assert isinstance(repr(combined), str)
+            assert "CombinedDrive" in repr(combined)
+            assert "Drive" in repr(combined)
+
+    def test_x(self):
+        for combined in self.combined_drives:
+            assert isinstance(combined.x, str)
+
+            assert combined.x in ["t", "iteration", "B_hysteresis"]
+
+        self.combined_drives[0].x = "mx"
+        with pytest.raises(ValueError):
+            self.combined_drives[0].x = "wrong"
+
+    def test_info(self):
+        for combined in self.combined_drives:
+            assert isinstance(combined.info, dict)
+            assert isinstance(combined.info["drive_numbers"], list)
+            assert combined.info["driver"] in [
+                "MinDriver",
+                "TimeDriver",
+                "HysteresisDriver",
+            ]
+
+    def test_m0(self):
+        for drive in self.combined_drives:
+            assert isinstance(drive.m0, df.Field)
+
+    def test_table(self):
+        for drive in self.combined_drives:
+            assert isinstance(drive.table, ut.Table)
+            assert drive.table.x == drive.x
+
+    def test_n(self):
+        assert self.combined_drives[0].n == 50
+        assert self.combined_drives[1].n == 15
+        assert self.combined_drives[2].n == 82
+
+    def test_getitem(self):
+        for i in range(self.combined_drives[1].n):
+            assert isinstance(self.combined_drives[1][i], df.Field)
```

Comment:
`assert all(isinstance(self.combined_drives[1][i], df.Field) for i in range(self.combined_drives[1].n))`